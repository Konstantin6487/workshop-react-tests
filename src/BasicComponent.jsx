import React from 'react';
import { Tabs, TabList } from 'react-tabs';

const BasicComponent = ({ renderTabsTitles, renderTabsContent }) => (
  <Tabs>
    <TabList data-test="tabs-container">{renderTabsTitles()}</TabList>
    {renderTabsContent()}
  </Tabs>
);

export default BasicComponent;
