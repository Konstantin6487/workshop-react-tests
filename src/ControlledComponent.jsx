import React, { Component } from 'react';
import {
  Tab,
  Tabs,
  TabList,
  TabPanel,
} from 'react-tabs';

export default class ControlledComponent extends Component {
  state = { tabIndex: 0 };

  handleSelect = (tabIndex) => {
    this.setState({ tabIndex });
  };

  render() {
    const { tabIndex } = this.state;
    return (
      <Tabs selectedIndex={tabIndex} onSelect={this.handleSelect}>
        <TabList>
          <Tab>Title 1</Tab>
          <Tab>Title 2</Tab>
        </TabList>
        <TabPanel />
        <TabPanel />
      </Tabs>
    );
  }
}
