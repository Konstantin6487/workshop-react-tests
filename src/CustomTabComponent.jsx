import React from 'react';

import {
  Tabs,
  TabList,
  Tab,
  TabPanel,
} from 'react-tabs';

const CustomTab = ({ children }) => (
  <Tab>
    <h1>{children}</h1>
  </Tab>
);

CustomTab.tabsRole = 'Tab'; // Required field to use your custom Tab

const CustomTabComponent = () => (
  <Tabs>
    <TabList>
      <CustomTab>Custom Tab 1</CustomTab>
      <CustomTab>Custom Tab 2</CustomTab>
    </TabList>
    <TabPanel>Panel 1</TabPanel>
    <TabPanel>Panel 2</TabPanel>
  </Tabs>
);

export default CustomTabComponent;
