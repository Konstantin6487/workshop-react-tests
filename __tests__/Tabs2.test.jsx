import React from 'react';
import BasicComponent from '../src/BasicComponent';
import ControlledComponent from '../src/ControlledComponent';
import CustomTabComponent from '../src/CustomTabComponent';

describe('enzyme', () => {
  test('<BasicComponent />', () => {
    const wrapper = mount(<BasicComponent />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('<ControlledComponent />', () => {
    const wrapper = mount(<ControlledComponent />);
    expect(wrapper.render()).toMatchSnapshot();

    wrapper.find('.react-tabs__tab').at(1).simulate('click');
    expect(wrapper.render()).toMatchSnapshot();

    wrapper.find('.react-tabs__tab').at(2).simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('<CustomTabComponent />', () => {
    const wrapper = mount(<CustomTabComponent />);
    expect(wrapper.render()).toMatchSnapshot();

    wrapper.find('.react-tabs__tab').at(1).simulate('click');
    expect(wrapper.render()).toMatchSnapshot();

    wrapper.find('.react-tabs__tab').at(2).simulate('click');
    expect(wrapper.render()).toMatchSnapshot();
  });
});
