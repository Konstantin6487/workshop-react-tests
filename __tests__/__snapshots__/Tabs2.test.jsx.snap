// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme <BasicComponent /> 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-0"
      role="tab"
      tabindex="0"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      id="react-tabs-2"
      role="tab"
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    <h2>
      Any content 1
    </h2>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
</div>
`;

exports[`enzyme <ControlledComponent /> 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-4"
      role="tab"
      tabindex="0"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      id="react-tabs-6"
      role="tab"
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-5"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-6"
    class="react-tabs__tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`enzyme <ControlledComponent /> 2`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-4"
      role="tab"
      tabindex="0"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      id="react-tabs-6"
      role="tab"
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-5"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-6"
    class="react-tabs__tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`enzyme <ControlledComponent /> 3`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      id="react-tabs-4"
      role="tab"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-6"
      role="tab"
      tabindex="0"
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-6"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`enzyme <CustomTabComponent /> 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      role="tab"
    >
      <h1>
        Custom Tab 1
      </h1>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      role="tab"
    >
      <h1>
        Custom Tab 2
      </h1>
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-8"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-9"
    role="tabpanel"
  >
    Panel 1
  </div>
  <div
    aria-labelledby="react-tabs-10"
    class="react-tabs__tab-panel"
    id="react-tabs-11"
    role="tabpanel"
  />
</div>
`;

exports[`enzyme <CustomTabComponent /> 2`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      role="tab"
    >
      <h1>
        Custom Tab 1
      </h1>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      role="tab"
    >
      <h1>
        Custom Tab 2
      </h1>
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-8"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-9"
    role="tabpanel"
  >
    Panel 1
  </div>
  <div
    aria-labelledby="react-tabs-10"
    class="react-tabs__tab-panel"
    id="react-tabs-11"
    role="tabpanel"
  />
</div>
`;

exports[`enzyme <CustomTabComponent /> 3`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      role="tab"
    >
      <h1>
        Custom Tab 1
      </h1>
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      role="tab"
    >
      <h1>
        Custom Tab 2
      </h1>
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-8"
    class="react-tabs__tab-panel"
    id="react-tabs-9"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-10"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-11"
    role="tabpanel"
  >
    Panel 2
  </div>
</div>
`;
